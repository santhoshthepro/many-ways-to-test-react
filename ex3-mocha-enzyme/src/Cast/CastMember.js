var React = require('react');

var CastMember = React.createClass({
  render: function() {
    var imageUrl = `/images/${this.props.member.thumbnailUrl}`;

    return (
      <div className="media">
        <div className="media-left">
          <a href="#">
            <img className="media-object" src={imageUrl} alt={this.props.member.name} />
          </a>
        </div>
        <div className="media-body">
          <h2 className="media-heading">{this.props.member.name}</h2>
          {this.props.member.bio}
        </div>
      </div>
    );
  }
});

module.exports = CastMember;

