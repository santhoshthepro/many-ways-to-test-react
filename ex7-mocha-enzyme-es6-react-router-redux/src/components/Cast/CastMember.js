import React, {PropTypes} from 'react';

const CastMember = ({ member }) => {
  const imageUrl = `/images/${member.thumbnailUrl}`;

  return (
    <div className="media">
      <div className="media-left">
        <a href="#">
          <img className="media-object" src={imageUrl} alt={member.name} />
        </a>
      </div>
      <div className="media-body">
        <h2 className="media-heading">{member.name}</h2>
        {member.bio}
      </div>
    </div>
  );
};

CastMember.propTypes = {
  member: PropTypes.object.isRequired
};

export default CastMember;
