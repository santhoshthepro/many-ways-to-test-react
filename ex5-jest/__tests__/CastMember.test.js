import React from 'react';
import renderer from 'react-test-renderer';
import CastMember from '../src/Cast/CastMember';

test('CastMember rendered static HTML', () => {
  const props = {
    member: {
      id: 3,
      name: 'Testy McTestFace',
      imageUrl: 'test-image.jpg',
      thumbnailUrl: 'test-image-small.jpg',
      bio: 'This is a test.'
    }
  };

  const component = renderer.create(
    <CastMember {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
