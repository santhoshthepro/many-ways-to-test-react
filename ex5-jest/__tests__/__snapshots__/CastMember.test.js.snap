exports[`test CastMember rendered static HTML 1`] = `
<div
  className="media">
  <div
    className="media-left">
    <a
      href="#">
      <img
        alt="Testy McTestFace"
        className="media-object"
        src="/images/test-image-small.jpg" />
    </a>
  </div>
  <div
    className="media-body">
    <h2
      className="media-heading">
      Testy McTestFace
    </h2>
    This is a test.
  </div>
</div>
`;
